components/places/src/storage/bookmarks.rs

```diff
@@ -121,6 +121,45 @@ pub enum BookmarkPosition {
     Append,
 }
 
+/// Helpers to deal with managing the position correctly.
+
+/// Updates the position of existing items so that the insertion of a child in
+/// the position specified leaves all siblings with the correct position.
+/// Returns the index the item should be inserted at.
+fn resolve_pos_for_insert(
+    db: &impl ConnExt,
+    pos: &BookmarkPosition,
```

Comment:
What if we had this function take `child: &InsertableItem, parent: &RawBookmark`?

Reply:
Good idea - I did roughly that.